[CSS] Animation - Add missing content and other edits

dipikabh opened the pull request:
<!-- 👀 Thanks for opening a PR! Read comments like this one to get your PR merged faster. -->
#### Summary
<!-- ✍️ In a sentence or two, describe your changes -->
This PR makes the following updates:

1. [animation-composition](https://developer.mozilla.org/en-US/docs/Web/CSS/animation-composition) page: Fixes a pending [review comment](https://github.com/mdn/content/pull/19848#discussion_r960161416)

2. [animation](https://developer.mozilla.org/en-US/docs/Web/CSS/animation) page:
  - Moves syntax box inside the "Syntax" section.
  - Changes syntax examples to indicate that the animation name comes at the end
  - Adds a syntax example to show more than one animation definition
  - Adds missing content for `<single-transition-easing-function>`
  - Rewrites a para for clarity
 
3. Other pages: Adds the list of related animation properties to "See also" section to aid in navigation

dipikabh committed:
Adds review suggestions

dipikabh committed:
Fixes syntax placement, adds missing content, and edits

dipikabh committed:
Adds links to other properties

estelle commented on files/en-us/web/css/animation/index.md:
```diff
@@ -44,27 +32,42 @@ This property is a shorthand for the following CSS properties:
+The order of time values within each animation definition is important: the first value that can be parsed as a {{cssxref("&lt;time&gt;")}} is assigned to the {{cssxref("animation-duration")}}, and the second one is assigned to {{cssxref("animation-delay")}}.
 
-The order within each animation definition is also important for distinguishing {{cssxref("animation-name")}} values from other keywords. When parsed, keywords that are valid for properties other than {{cssxref("animation-name")}}, and whose values were not found earlier in the shorthand, must be accepted for those properties rather than for {{cssxref("animation-name")}}. Furthermore, when serial ...
+The order of other values within each animation definition is also important for distinguishing an {{cssxref("animation-name")}} value from other values. If values in the animation definition can be parsed for properties other than `animation-name` and if those properties are assigned values for the first time in the shorthand, then they must be accepted for those properties and not for `animatio ...
```
```suggestion
The order of other values within each animation definition is also important for distinguishing an {{cssxref("animation-name")}} value from other values. If values in the animation definition can be parsed for properties other than `animation-name` and if those properties are assigned values for the first time in the shorthand, then they must be accepted for those properties and not for `animation-name`. Furthermore, when serialized, default values of other properties must be output in at least the cases necessary to distinguish an `animation-name` that could be a value of another property and may be output in additional cases.

While neither an `animation-name` nor `animation-duration` is required to be declared within an `animation` shorthand property, both are needed for an animation to occur.
```

estelle replied:
The example on this page animates a box-model property (margin-left) which is a bad practice. We should change that to a background-position animation or a transform.

estelle replied:
looks good. just these two comments.

dipikabh committed:
Fixes review comments

dipikabh committed:
Fixes trailing space

dipikabh commented:
Hi @estelle, thanks for the review! I've incorporated your comments.

1. Corresponding to your first suggestion, I also updated the descriptions in the `animation-name` and `animation-duration` pages. The more detailed description for animation visibility when duration is 0s is on the `animation-duration` page.
2. The example on the `animation` page had to be completely overhauled. That took a while.

Let me know what you think.

estelle commented on files/en-us/web/css/animation-name/index.md:
```diff
@@ -11,7 +11,7 @@ browser-compat: css.properties.animation-name
 {{CSSRef}}
 
-The **`animation-name`** [CSS](/en-US/docs/Web/CSS) property specifies the names of one or more {{cssxref("@keyframes")}} at-rules describing the animation or animations to apply to the element.
+The **`animation-name`** [CSS](/en-US/docs/Web/CSS) property specifies the names of one or more {{cssxref("@keyframes")}} at-rules describing the animation or animations to apply to the element. If the specified name does not match any `@keyframe` at-rule, no properties are animated.
```
your grammar is better than mine. i might be wrong about the s.

dipikabh replied:
I modified the first sentence a bit and added a new sentence to state the 'comma-separated' bit. I've retained the original sentence "If the specified name..." because to stay consistent with the first sentence.

estelle reviewed (commented):
just a few comments. looking good.

dipikabh committed:
Fixes review comments

dipikabh committed:
Removes unused gif file

estelle commented on files/en-us/web/css/animation-duration/index.md:
```diff
@@ -40,7 +40,12 @@ animation-duration: unset;
-  - : The time that an animation takes to complete one cycle. This may be specified in either seconds (`s`) or milliseconds (`ms`). The value must be positive or zero and the unit is required. A value of `0s`, which is the default value, indicates that no animation should occur.
+  - : The time that an animation takes to complete one cycle. This may be specified in either seconds (`s`) or milliseconds (`ms`). The value must be positive or zero and the unit is required.
+
+    If no value is provided, the default value of `0s` is used, in which case the animation still executes (the animation start and end events are fired). Whether or not the animation will be visible when duration is `0s` will depend on the value of [`animation-fill-mode`](/en-US/docs/Web/CSS/animation-fill-mode), as explained below:
```
```suggestion
    If no value is provided, the default value of `0s` is used, in which case the animation still executes (the [`animationStart`](/en-US/docs/Web/API/Element/animationstart_event) and [`animationEnd`](/en-US/docs/Web/API/Element/animationend_event) events are fired). Whether or not the animation will be visible when the duration is `0s` will depend on the value of [`animation-fill-mode`](/en-US/docs/Web/CSS/animation-fill-mode), as explained below:
```

estelle commented on files/en-us/web/css/animation/index.md:
```diff
@@ -75,11 +74,21 @@ The order within each animation definition is also important for distinguishing
+
+The order of time values within each animation definition is important: the first value that can be parsed as a {{cssxref("&lt;time&gt;")}} is assigned to the {{cssxref("animation-duration")}}, and the second one is assigned to {{cssxref("animation-delay")}}.
+
+The order of other values within each animation definition is also important for distinguishing an {{cssxref("animation-name")}} value from other values. If values in the animation definition can be parsed for properties other than `animation-name` and if those properties are assigned values for the first time in the shorthand, then they must be accepted for those properties and not for `animatio ...
```
```suggestion
The order of other values within each animation definition is also important for distinguishing an {{cssxref("animation-name")}} value from other values. If values in the `animation` shorthand can be parsed as a  value for an animation property other than `animation-name` then the value will be applied to that property and not as the `animation-name`. For this reason, some recommend including the `animation-name` value as the last component when using the `animation` shorthand. This holds true for all `animation-name` values when using the `animation` shorthand to include multiple comma-separated animations.
```

estelle replied:
I tried to 3rd-grade-reading-level this.

dipikabh replied:
The improvements look good. Thanks. I have tweaked the last sentence a bit.

estelle commented on files/en-us/web/css/animation/index.md:
```diff
@@ -75,11 +74,21 @@ The order within each animation definition is also important for distinguishing
+
+An `animation-name` value is not required to be declared in the `animation` shorthand property. If no name exists, there is no animation to apply on any of the properties.
+
+When the `animation-duration` value is omitted from the `animation` shorthand property, the value for this property defaults to `0s`. In this case, the animation will still occur (the animation start and end events will be fired) but no animation will be visible.
```
```suggestion
When the `animation-duration` value is omitted from the `animation` shorthand property, the value for this property defaults to `0s`. In this case, the animation will still occur (the [`animationStart`](/en-US/docs/Web/API/Element/animationstart_event) and [`animationEnd`](/en-US/docs/Web/API/Element/animationend_event) events will be fired) but no animation will be visible.
```

do we want to link to the section in animation-duration?

dipikabh replied:
`animation-duration` is linked just a few paragraphs above.

estelle commented on files/en-us/web/css/animation/index.md:
```diff
@@ -75,11 +74,21 @@ The order within each animation definition is also important for distinguishing
 Blinking and flashing animation can be problematic for people with cognitive concerns such as Attention Deficit Hyperactivity Disorder (ADHD). Additionally, certain kinds of motion can be a trigger for Vestibular disorders, epilepsy, and migraine and Scotopic sensitivity.
 
-Consider providing a mechanism for pausing or disabling animation, as well as using the [Reduced Motion Media Query](/en-US/docs/Web/CSS/@media/prefers-reduced-motion) to create a complimentary experience for users who have expressed a preference for no animated experiences.
+Consider providing a mechanism for pausing or disabling animation as well as using the [Reduced Motion Media Query](/en-US/docs/Web/CSS/@media/prefers-reduced-motion) to create a complimentary experience for users who have expressed a preference for no animated experiences.
```
```suggestion
Consider providing a mechanism for pausing or disabling animation as well as using the [Reduced Motion Media Query](/en-US/docs/Web/CSS/@media/prefers-reduced-motion) to create a complimentary experience for users who have expressed a preference for minimal animated experiences.
```
reduced motion doesn't mean no motion.

dipikabh committed:
Fixes review comments

estelle reviewed (approved):
yay!
